src/SynchronizesWithMauticTrait.php

```diff
@@ -2,21 +2,22 @@
 
 namespace Swis\Laravel\Mautic;
 
+use Illuminate\Database\Eloquent\Model;
 use Swis\Laravel\Mautic\Jobs\DeleteModelFromMautic;
 use Swis\Laravel\Mautic\Jobs\PersistModelInMautic;
 
 trait SynchronizesWithMauticTrait
 {
     public static function bootSynchronizesWithMauticTrait(): void
     {
-        static::saved(function ($model) {
+        static::saved(function (Model $model) {
             if ($model instanceof SynchronizesWithMautic) {
-                PersistModelInMautic::dispatch($model);
+                dispatch($model->getOnSavedJob());
             }
         });
-        static::deleted(function ($model) {
+        static::deleted(function (Model $model) {
             if ($model instanceof SynchronizesWithMautic) {
-                DeleteModelFromMautic::dispatch($model);
+                dispatch($model->getOnDeletedJob());
```

Comment:
What if someone doesn't want to delete the model from Mautic and returns something like `null`?

Reply:
Good point. I will add a null check.